integration.md

```diff
@@ -0,0 +1,130 @@
+top-level browsing contexts only, and are not available in cross-origin child
+frames, for security. These kinds of features should use a default allowlist of
+`["self"]` to provide this behaviour. In this case, Feature Policy also grants
+site authors the ability to selectively enable the feature in other origins,
+but the default behaviour is to disable it.
+* A default allowlist of `[]` should be restricted to new features, as a
+Feature Policy header is *required* in order to enable it in any document.
+
+## Considerations for behavior when disallowed
+
+It is important for feature authors to carefully consider the mechanism by
+which their feature is disabled when disallowed in a document, for ease of
+developer feature detection, as well as for compatibility with existing web
+content.
+
+Existing web platform features which already have some mechanism for reporting
+errors, or signalling that the user chose not to grant permission, should use
+those existing mechanisms when disabled.
+
+New features should attempt to fail in a way which allows authors to detect
+that the feature is disabled.
+
+> Example:
+>
+> The fullscreen API has existed on the web for some time, and is being
+> implemented as a policy-controlled feature. Developers have always had to
+> contend with the fact that fullscreen may not be allowed, and the API has a
+> method of signalling that.
+>
+> As a policy-controlled feature, the fullscreen API can specify that when
+> disabled, `document.fullscreenEnabled` must return `false`, and
+> `Element.requestFullscreen()` must fire a `fullscreenerror` event, and return
+> a promise which rejects with a `TypeError` exception.
+
+If an existing feature has no mechanism for reporting failure, or has never been
+allowed to fail in the past, then it may be dangerous to turn it into a
+policy-controlled feature, where it may start to fail on pages that were always
+able to assume that it would not.
+
+In this case, features can take advantage of the concept of
+[feature-policy-awareness](https://wicg.github.io/feature-policy/#feature-policy-aware)
+to define different behaviour when disabled in a document written to be aware of
+Feature Policy (and hence the possiblity that the API could fail) and in a
+document written without that knowledge.
+
+> Example:
+>
+> document.write API has existed on the web since roughly the dawn of
+> JavaScript, and has no failure modes. This means that if a page contains a
+> script like this:
+>
+>     <html>
+>       <head>
+>         <script>document.write('<script src="important-security-feature.js"></scr'+'ipt>')</script>
+>         ...
+>
+> then an attacker could embed the site in a frame with the `document.write`
+> feature disabled, and bypass the critical security script.
+>
+> To avoid this, if disallowed in a feature-policy-aware document, calling
+> `document.write()` will fire a `SecurityError` exception on the `document`
```

Comment:
Interesting. I would be worried that people would start using feature policy without realizing this implication and open themselves up to security problems.